### PR TITLE
MRC-71-Implement acceptance test for MRC RestAPI 

### PR DESCRIPTION
**Pull request by Sara-raj**
image

### CODE REVIEW

File: src/test/resources/features/MarginalReliefCalculator.feature [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Feature:  Basic MR calculation for single financial year with MR rate
+
+  Scenario:  MR-68 AC1  1 COMPLETE YEAR + PROFITS LIES WITHIN MR THRESHOLDS
+    When a request is made to GET response from MRC service for /calculate with query params 2023-04-01, 2024-03-31, 100000
```
Comment (girishkamat) [inline src/test/resources/features/MarginalReliefCalculator.feature:4]: For readability, can this be rephrased?

When performing a calculate request with accountingPeriodStart as 2023-04-01, accountingPeriodEnd as 2024-03-31, profit as 100000

File: src/test/resources/features/MarginalReliefCalculator.feature [hunk 2]
```diff
@@ -0,0 +1,44 @@
+Feature:  Basic MR calculation for single financial year with MR rate
```
Comment (girishkamat) [inline src/test/resources/features/MarginalReliefCalculator.feature:1]: The file could be named as MarginalReliefCalculator-calculate.feature. There is will be other endpoint that check if associate companies should be requested, if yes should associated companies be requested a single value or for both years etc

File: src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala [hunk 3]
```diff
@@ -1,30 +1,108 @@
...
   }
 
   Then("the MRC response code should be (.*)") { expectedCode: String =>
     val response: StandaloneWSResponse = ScenarioContext.get("response")
-    response.body.equals(expectedCode)
+    response.body.contains(expectedCode)
   }
 
   And("the MRC hello world response body should be (.*)") { message: String =>
     val response: StandaloneWSResponse = ScenarioContext.get("response")
```
Comment (girishkamat) [inline src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala:44]: is this required - "the MRC hello world response body should be (.*)"

File: src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala [hunk 4]
```diff
@@ -1,30 +1,108 @@
...
   }
+  And("MRC response contains") { (dataTable: DataTable) =>
+    val asMapTransposed                = dataTable.transpose().asMap(classOf[String], classOf[String])
+    val response: StandaloneWSResponse = ScenarioContext.get("response")
+    val responseBody                   = Json.parse(response.body).as[SingleYearCalculationSummary]
 
-}
\ No newline at end of file
+    if (asMapTransposed.containsKey("effectiveTaxRateBeforeMR")) {
+      responseBody.effectiveTaxRateBeforeMR.toString shouldBe asMapTransposed.get("effectiveTaxRateBeforeMR").toString
```
Comment (girishkamat) [inline src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala:53]: shouldnt this fail if effectiveTaxRateBeforeMR (or other attributes)  are missing in json?

File: project/plugins.sbt [hunk 5]
```diff
@@ -3,5 +3,6 @@ resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts
   Resolver.ivyStylePatterns
 )
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.6.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "2.14.0")
```
[the remark this answers is not in the document]
Comment (Sara-raj) [reply project/plugins.sbt:6]: yes. there was an issue when I try to run my .sh script.

File: src/test/resources/features/MarginalReliefCalculator.feature (hunk 1 above)
Comment (Sara-raj) [reply src/test/resources/features/MarginalReliefCalculator.feature:4]: done

File: src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala (hunk 3 above)
Comment (Sara-raj) [reply src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala:44]: my bad. removed the code

File: src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala (hunk 4 above)
Comment (Sara-raj) [reply src/test/scala/uk/gov/hmrc/test/api/cucumber/stepdefs/MarginalReliefCalculatorSteps.scala:53]: yes it will

**Review by girishkamat — APPROVED** [review]

**Pull request merged**